rosbag2_storage/include/rosbag2_storage/storage_interfaces/base_read_interface.hpp

```diff
@@ -62,8 +62,10 @@ class ROSBAG2_STORAGE_PUBLIC BaseReadInterface
   ///   This affects the outcome of has_next and read_next.
   ///   Note that when setting to reverse order, this will not change the read head, so user
   ///   must first seek() to the end in order to read messages from the end.
+  ///   Also note that set_read_order should only be called after open() has been called.
```

Comment:
Maybe specify that `set_read_order` should always return false before `open`?

Reply:
i think that might mislead callers - there's a difference between "this read order is not supported" and "you have called these functions in the wrong order, go change your code".

Reply:
So `if !open: throw; elif open && unsupported: return false`? 

I just want the exact behavior expectation to be specified here. If ^ then should add `/// \throws` to docstring